### DISCUSSION TITLE
A handle to turn number rounding off for panel visualization

### DISCUSSION THREAD

**Discussion by ptimofee** [grafana/grafana#61809] [Dashboards, Panels & Visualizations]
There is an option in most of panel/visualization types that allows you to show a specific number of decimal digits after the point in a metric value: Standard Options -> Decimals.
Turns out it not only shows those digits, but also rounds the result (grafana 9.3.2).

Can we have a checkbox somewhere in this Standard Options ui block to turn rounding off?

The use case is really simple: SLO/SLA calculation/representation.
Typical SLO/SLA representation is a number of percents from 0 to 100 with up to 3 decimal digits, for example 99.999%, 99.99% or 99.9%.
Or you may have other numbers as an agreement with a customer.

And let's say you have some comlicated prometheus query, that returns SLO result as 99.99976896976897.
You want to see 99.999 in your Stat panel so you go ahead and set Standard Options -> Decimals to 3. Surprisingly you'll see 100 visualized.

Or as another example: a customer wants SLA of 99.0%.
So 98.99976896976897 returned by prometheus will give you 99.000 with Standard Options -> Decimals set to 3. This is not good.

Setting Standard Options -> Decimals to 4 (or any higher number) only works around the problem. You may have SLO result 99.99996896976897 and setting of 4 decimals will round it. Etc.

SLA is a crucial part of a service availability evaluation and number rounding gives you wrong estimate.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82858. Feel free to continue the discussion around this there. Thank you!
